low-level/solution/lr2/support_bundle/sspl_support_bundle

```diff
@@ -0,0 +1,388 @@
+            self.iem_log_dir: os.path.join(file_dir, "iems/")
+        }
+        EXCLUDE_FILES = [sspl_const.SB_DATA_PATH]
+        tar_location = os.path.normpath(self.tmp_bundle_path+"/"+file_name)
+        try:
+            with tarfile.open(tar_location, "w:gz") as tar:
+                for key, value in tarfile_data.items():
+                    if os.path.exists(key):
+                        try:
+                            tar.add(key, arcname=value,
+                                    exclude=lambda x: x in EXCLUDE_FILES)
+                        except IOError as err:
+                            logger.error("Unable to include {0} logs "
+                            "with an error {1}".format(key, err))
+            return tar_location
+        except (OSError, tarfile.TarError) as err:
+            msg = "Facing issues while creating sspl "
+            "support bundle: {0}".format(err)
+            self.__clear_tmp_files()
+            raise SupportBundleError(1, msg)
+
+    def __clear_tmp_files(self):
+        """Clean temporary files created by the support bundle."""
+        if os.path.exists(sspl_const.SSPL_SB_TMP):
+            shutil.rmtree(sspl_const.SSPL_SB_TMP)
+
+    @staticmethod
+    def parse_arguments():
+        parser = argparse.ArgumentParser(description="SSPL Support Bundle")
+        parser.add_argument('bundle_id',
+                            help='Unique bundle id')
+        parser.add_argument('target_path',
+                            help='Path to store the created bundle',
+                            nargs='?',
+                            default="/var/seagate/cortx/support_bundle/")
+        parser.add_argument('-noencl',
+                            action='store_true',
+                            help='Exclude enclosure logs')
+        parser.add_argument('-console',
+                            action='store_true',
+                            help='Print logs on console')
+        args = parser.parse_args()
+        return args
+
+
+def initialize_logging(parser):
+    # set Logging Handlers
+    _logger = logging.getLogger('sspl_sb')
+    logging_level = Conf.get(
+        SSPL_CONF, f"{'SYSTEM_INFORMATION'}>{'log_level'}", "INFO")
+    _logger.setLevel(logging_level)
+    handler = logging.handlers.SysLogHandler(
+        address=(sspl_const.SYSLOG_HOST, sspl_const.SYSLOG_PORT))
+    syslog_format = "%(name)s[%(process)d]: " \
+                    "%(levelname)s %(message)s (%(filename)s:%(lineno)d)"
+    formatter = logging.Formatter(syslog_format)
+    handler.setFormatter(formatter)
+    _logger.addHandler(handler)
+
+    # Add console handler
+    if "-console" in parser:
+        console = logging.StreamHandler(sys.stdout)
+        console.setFormatter(formatter)
+        _logger.addHandler(console)
+    return _logger
+
+
+logger = initialize_logging(sys.argv)
+
+
+def main():
+    bundle = SupportBundle()
+    args = bundle.parse_arguments()
+    bundle.generate(args)
```

Comment:
where do we check if support bundle process is already running?

Reply:
As @satish-seagate replied.